SnapsInAZfs/ConfigConsole/SnapsInAZfsConfigConsole.cs

```diff
@@ -249,20 +221,11 @@ private void ShowGlobalConfigurationWindow( )
 
         _globalConfigurationWindow ??= new( );
         Add( _globalConfigurationWindow );
-        if ( ShowChild( _globalConfigurationWindow ) )
```

Comment:
> A version of `Add` that returns a boolean would be great, here.
> 
Depends, you may want to add the same view more than once on the same superview.

> I liked that `ShowChild` returned a boolean, because it guarded against unforeseen failures.
> 
This is different because the Application.Top is a MdiContainer and it returns true if the Child you want to show was previously the current and false if it's already the current. If you preferred this behavior you must set the main top with `IsMdiContainer = true` and use `Application.Run<myChildWindow>()`, but you must set the all the configuration, like the view you want to have focus, etc, before this call.

> Just a suggestion, if V2 uses the same general flow.

Not worth because of the reason I said before.

> Alternatively, a more standard approach for a boolean method that is guaranteed not to throw an exception would be the `TryX` pattern.

The reason of the error was because of the absent of `Application.Init` that I'll investigate.

Reply:
> The reason of the error was because of the absent of `Application.Init` that I'll investigate.

Sorry I was wrong here. The reason was the use of `IsMdiContainer = true` on a child window instead on the parent window and also using the `Add(myChildWindow)` instead of `Application.Run<myChildWindow>()`.

Reply:
> > The reason of the error was because of the absent of `Application.Init` that I'll investigate.
> 
> Sorry I was wrong here. The reason was the use of `IsMdiContainer = true` on a child window instead on the parent window and also using the `Add(myChildWindow)` instead of `Application.Run<myChildWindow>()`.

I wasn't using IsMdiContainer on any child windows, though. The only one that was setting it was SnapsInAZfsConfigConsole, which is the parent.

That ConfigConsole class (Somewhat confusingly named, I know) is just a static class that sets it up. I have plans to refactor that anyway.

When I tried using Application.Run for a child window, it had the undesired behavior of terminating the whole GUI when RequestStop was called from that child, rather than only closing the child itself.

Reply:
> > A version of `Add` that returns a boolean would be great, here.
> 
> Depends, you may want to add the same view more than once on the same superview.

Makes sense. Of course, in my use case, they are singletons, but that makes sense for the general case. Still, returning a boolean that indicates success or failure of the operation could be helpful, or returning the reference to the created child itself.

> > I liked that `ShowChild` returned a boolean, because it guarded against unforeseen failures.
> 
> This is different because the Application.Top is a MdiContainer and it returns true if the Child you want to show was previously the current and false if it's already the current. If you preferred this behavior you must set the main top with `IsMdiContainer = true` and use `Application.Run<myChildWindow>()`, but you must set the all the configuration, like the view you want to have focus, etc, before this call. I don't know if the `TerminalGuiDesigner` is generating this properly when `IsMdiContainer = true`.

Ah. That's an interesting implementation detail that isn't clear from the existing V1 documentation. The Mdi case isn't well documented for V1, so I mostly had to just throw stuff at the wall to see what stuck. 😅

Reply:
To close a child you must use myChild.RequestStop on Mdi.

Reply:
Ah, so you mean the parent is responsible for calling it?

Why would it matter, though? Ultimately, the same function is being called on the same object, so it's unclear to me.

Reply:
> I wasn't using IsMdiContainer on any child windows, though. The only one that was setting it was SnapsInAZfsConfigConsole, which is the parent.

Sorry for my mistake. You are right. What was causing the malfunction was the call to `ShowChild` method when you used the `Add` method instead of the `Application.Run`.

>> To close a child you must use myChild.RequestStop on Mdi.

> Ah, so you mean the parent is responsible for calling it?
> 
Yes, on a Mdi the parent is who handle his childs. The parent is the `Application.MdiTop`.

> Why would it matter, though? Ultimately, the same function is being called on the same object, so it's unclear to me.

I understand your doubts. On a MdiTop you add MdiChilds as needed by using Application.Run<MyChildClass> or  Application.Run(myChildInstance). A `MdiChild` is a non-modal toplevel. So, if a `MdiChild` has focus it's the current toplevel (Application.Current), but also MdiTop can be the current top in the case where a subview is focused, for e.g. the MenuBar. If you use the `Applcation.RequestStop()` when the MdiTop is the `Application.Current` the app will be closed and exit. That why it's better use `myChild.RequestStop()` or parent.RequestStop(myChild) and the MdiTop will only close the myChild instance and not will exit your app. If you want to hide a child you can use `myChild.HideChild()` and to show again use `myChild.ShowChild()`.

Reply:
> Makes sense. Of course, in my use case, they are singletons, but that makes sense for the general case. Still, returning a boolean that indicates success or failure of the operation could be helpful, or returning the reference to the created child itself.

For the `Add` method I think there is no sense to return something because it will returning always true. Also returning a reference is unnecessary because you already have the reference. When you do superview.Add(subview) you already have the subview reference and when you do superview.Add(subview1, subview2) you already have subview1and subview2 references. Notice that you can't add a MdiChild with the Add method, but only the MdiChild.SubViews.
For the `ShowChild` it already returns true if it wasn't previously current and now it is current to allow you do some actions like setting some values or focused some view, etc. If it returns false it was already the current and you may not do nothing because the user is currently handle it, or do whatever you want.

Reply:
> Ah. That's an interesting implementation detail that isn't clear from the existing V1 documentation. The Mdi case isn't well documented for V1, so I mostly had to just throw stuff at the wall to see what stuck. 😅

Yes, you are right. Unfortunately I'm not so good to deal with documentation. If you want to provide some help with that on the `Terminal.Gui` I really appreciate 😃